extensions/azurecore/src/azureResource/utils.ts

```diff
@@ -288,18 +288,11 @@ export async function getSelectedSubscriptions(appContext: AppContext, account?:
 	return result;
 }
 
-enum HttpRequestType {
-	GET,
-	POST,
-	PUT,
-	DELETE
-}
-
 /**
- * Make a HTTP request to Azure REST apis.
+ * Make an HTTP request to Azure REST apis.
  */
-export async function makeHttpRequest(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, ignoreErrors: boolean = false, url: string, requestType: HttpRequestType, requestBody?: any): Promise<HttpRequestResult> {
-	const result: HttpRequestResult = { response: {}, errors: [] };
+export async function makeHttpRequest(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, path: string, requestType: HttpRequestMethod, requestBody?: any, ignoreErrors: boolean = false, host?: string): Promise<AzureRestResponse> {
```

Comment:
I don't think host should be optional - there isn't really a use case for that currently and I'd rather callers know exactly what endpoint they're connecting to for now

And a couple other notes here for future things : 

1. If it was then I would suggest you use the default value in the parameter instead - that makes it much clearer what the default is
2. You should always be documenting in the comments what defaults are if there are any